### PR TITLE
convert bool to string

### PR DESCRIPTION
**Pull request by xfate123**

### CODE REVIEW

**Comment by xfate123** [comment]
@terrytangyuan Hi, terry. This pr is related to issue #29. I don't know whether I understand your requirement correctly. I also don't know how to fix these failed check.
Thank you again

**Comment by terrytangyuan** [comment]
Please take a look at the contributing guide here to fix the the build: https://github.com/couler-proj/couler/blob/master/CONTRIBUTING.md

File: couler/core/templates/job.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(
         self.args = args
         self.action = action
         self.manifest = manifest
-        self.set_owner_reference = "true" if set_owner_reference else "false"
+        self.set_owner_reference = utils.bool_to_str(set_owner_reference)
```
Comment (terrytangyuan) [inline couler/core/templates/job.py:40]: Could you check if there are other places that require this and update those as well?

File: couler/core/utils.py [hunk 2]
```diff
@@ -255,3 +255,8 @@ def _get_uuid():
 
     """use uuid4 to gen ascii uuid, length=8"""
     return "".join(str(uuid.uuid4()).split("-"))[:8]
+def bool_to_str(b):
```
Comment (terrytangyuan) [inline couler/core/utils.py:258]: Add a newline?

File: couler/core/utils.py [hunk 3]
```diff
@@ -255,3 +255,8 @@ def _get_uuid():
...
+    if b:
```
Comment (terrytangyuan) [inline couler/core/utils.py:259]: Avoid single letter variable name. Perhaps renaming to bool_val? Also need to check whether this argument is boolean.

File: couler/core/utils.py [hunk 4]
```diff
@@ -255,3 +255,8 @@ def _get_uuid():
...
+        return 'true'
```
Comment (terrytangyuan) [inline couler/core/utils.py:260]: Optimize to use one-liner?

Comment (xfate123) [reply couler/core/utils.py]: @terrytangyuan  Thank you so much for your help.  These are few changes I have made. Feel free to give me any advice.
I still have trouble to set up environment for the test, I will figure it out tomorrow and check out any other places that need this util function as well.

File: couler/core/templates/job.py (hunk 1 above)
Comment (linkerzhang) [reply couler/core/templates/job.py:40]: this looks to be the only one place in the code base.

**Review by linkerzhang — APPROVED** [review]

**Comment by terrytangyuan** [comment]
@xfate123 CI is failing in sanity checks. Please take a look at the instruction here: https://github.com/couler-proj/couler/blob/master/CONTRIBUTING.md#run-sanity-checks

Other than that, this is good to merge once that's fixed.

File: couler/core/templates/job.py (hunk 1 above)
Comment (terrytangyuan) [reply couler/core/templates/job.py:40]: Thanks for checking. It can probably be reused in our internal extensions.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**